loopin.go

```diff
@@ -135,17 +135,55 @@ func newLoopInSwap(globalCtx context.Context, cfg *swapConfig,
 		return nil, err
 	}
 
+	// Create the probe invoice in lnd. Derive the payment hash
+	// deterministically from the swap hash in such a way that the server
+	// can be sure that we don't know the preimage.
+	probeHash := lntypes.Hash(sha256.Sum256(swapHash[:]))
+	probeHash[0] ^= 1
+
+	log.Infof("Creating probe invoice %v", probeHash)
+	probeInvoice, err := cfg.lnd.Invoices.AddHoldInvoice(
+		globalCtx, &invoicesrpc.AddInvoiceData{
+			Hash:   &probeHash,
+			Value:  lnwire.NewMSatFromSatoshis(swapInvoiceAmt),
+			Memo:   "probe",
+			Expiry: 3600 * 24 * 365,
+		},
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create a cancellable context that is used for monitoring the probe.
```

Comment:
nit: cancelable

Reply:
Using that spelling on the rpc is bad enough already.